core/src/main/java/hudson/model/Run.java

```diff
@@ -2283,7 +2293,6 @@ public EnvVars getEnvironment() throws IOException, InterruptedException {
         Node n = c==null ? null : c.getNode();
 
         EnvVars env = getParent().getEnvironment(n,listener);
-        env.putAll(getCharacteristicEnvVars());
 
         // apply them in a reverse order so that higher ordinal ones can modify values added by lower ordinal ones
         for (EnvironmentContributor ec : EnvironmentContributor.all().reverseView())
```

Comment:
`!(this instanceof AbstractBuild)` below is not required anymore, because `AbstractBuild` does not consult with `super` anymore

Reply:
Good point, it was likely needed at the time this work started. Done